lib/Migration/Version1040Date20200422142920.php

```diff
@@ -111,6 +111,13 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
 			'mail_msg_mb_uid_idx'
 		);
 		$messagesTable->addIndex(['sent_at'], 'mail_msg_sent_idx');
+		// mail_messages_msgid_idx was added later and may not exist until optional indices are created
+		$messagesTable->addIndex(
+			['message_id'],
+			'mail_messages_msgid_idx',
+			[],
+			['lengths' => [128]],
+		);
```

Comment:
This won't end up as an index in the table at new instance setup because we remove all indices in the next migration.

https://github.com/nextcloud/mail/blob/f50605daaa5e21fde1e46abd645b22589b628276/lib/Migration/Version1130Date20220412111833.php#L160-L165

I'm not sure why we do this, but it will probably lead to further issues when editing indices in former migrations. Already happened with `mail_msg_sent_idx` on `sent_at` which wasn't add again and the unique index `mail_msg_mb_uid_idx` on `['uid','mailbox_id']` which was removed but not added again right away. It was [added again](https://github.com/nextcloud/mail/pull/6509) after some time as a non-unique index `mail_messages_mb_id_uid` when perf issues occurred.